woocommerce/woocommerce
# [Accessibility] Add visual label to sort by select on Shop page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



This PR adds a setting to the Catalog Sorting block that allows editors to enable a visual label to the component (Sort by).
**Catalog Sorting on the frontend:**
![image](https://github.com/user-attachments/assets/77bcaff2-acee-4726-bfb0-456a63680229)

![image](https://github.com/user-attachments/assets/c6e39c10-f8cf-4889-b673-6374210a2e3b)




Closes #51930 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install a block-based theme.
2. Go to Appearance > Editor > Templates > WooCommerce > Product Catalog.
3. Select the "Catalog Sorting" block.
4. Verify the "Show visual label" setting is visible.
    ![image](https://github.com/user-attachments/assets/df61a91d-a1b6-4b9e-9b9f-898d0ef8d7d1)
5. Verify that:
    - If the setting is off, the "Sort by" label is not visible before the selector, meaning no changes should be applied to the block.
    - Go to the Shop on the frontend and verify that nothing changed in the behavior of the sorting select compared to what it was before this PR.
      ![image](https://github.com/user-attachments/assets/48a2cb5e-6b7d-4391-824b-badff9cfbe8f)
    - If the setting is on, there is a `<label>` tag before the selector that reads "Sort by".
    - Go to the Shop page on the frontend.
    - Verify the "Sort by" select input has a separate label that reads "Sort by".
    - Verify the option values don't have the "Sort by" text in the beginning.
    - Turn on the screen reader and verify the "Sort by" label is announced.
      ![image](https://github.com/user-attachments/assets/15f91964-9b43-4e77-a1cf-b5aff11ba0ca)
6. **HEADS-UP**: The setting doesn't affect Classic themes.




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
